### PR TITLE
Pairs Trading + Eigenportfolio + Signals

### PR DESCRIPTION
**Pull request by akwon31**
# Description

Changed all class structures and docs for Pairs Trading and Eigenportfolio. The stationarity + cointegration components are the same as last time, so that section can be skipped for review.

Edit: I finished signals as well and merged them into here.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:
* Operating system: Mac
* IDE used: PyCharm

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mlfinlab/statistical_arbitrage/signals.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        # Change nan to 0.
+        res = np.nan_to_num((data - np.mean(data, axis=0)) / np.std(data, axis=0))
+    return res
+
+
+def calc_ou_process(data):
+    """
+    Calculates the s-score, derived from the Ornstein-Uhlenbeck process, for the given data.
```
Comment (Jackal08) [inline mlfinlab/statistical_arbitrage/signals.py:27]: Lets link to an example so the reader can learn more about the S-Score.

**Comment by rspadim** [comment]
Just be carefull if we need to optimize code with many classes/objects

A plain function in python normally runs faster than many objects / classes, this is true when using pypi too

File: docs/source/statistical_arbitrage/signals.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+The higher the number, the longer it will take to revert back to its projected mean.
+
+Z-Score
+#######
+
+Z-Score is also another popular method of measuring the distance of the measurement from the mean.
+It has an underlying assumption that the distribution is Gaussian, which may not necessarily be
+true for many of the financial data available to us. Nonetheless, it is often used as the simple
+but powerful equation indicates the magnitude of the deviation from the mean.
```
Comment (PanPip) [inline docs/source/statistical_arbitrage/signals.rst:70]: Probably `equation that indicates`

**Comment by Jackal08** [comment]
Pls open the PR in Premium Repo.

**Pull request closed without merging**